hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieSparkSqlWriter.scala

```diff
@@ -415,6 +415,7 @@ object HoodieSparkSqlWriter {
 
         val allowAutoEvolutionColumnDrop = opts.getOrDefault(HoodieWriteConfig.SCHEMA_ALLOW_AUTO_EVOLUTION_COLUMN_DROP.key,
           HoodieWriteConfig.SCHEMA_ALLOW_AUTO_EVOLUTION_COLUMN_DROP.defaultValue).toBoolean
+        val mergeIntoWrites = opts.getOrDefault(DataSourceWriteOptions.MERGE_INTO_WRITES, DataSourceWriteOptions.DEFAULT_MERGE_INTO_WRITES).toBoolean
```

Comment:
we need to infer this when user is using merge into. it's gonna be pretty bad exp if someone uses merge into and still need to set the config = true.

Reply:
look into `org.apache.spark.sql.hudi.command.MergeIntoHoodieTableCommand#buildMergeIntoConfig`

Reply:
ok saw it's there in the files below